Pet-Clinic-Frontend/src/pages/Pets.tsx

```diff
@@ -3,23 +3,30 @@ import { AddNewPetForm } from "../components/AddNewPetForm"
 import { PetsList } from "../components/PetsList"
 
 export const Pets = () => {
-  const [renderForm, setShowForm] = useState(false)
+  const [renderForm, setRenderForm] = useState(false)
+  const [error, setError] = useState<string | null>(null)
 
   const showForm = () => {
-    setShowForm(!renderForm)
+    setRenderForm(!renderForm)
```

Comment:
```suggestion
    setRenderForm(currentState => !currentState)
```

you can use the function-styled setter, which is getting a function instead of a value, just overwriting the previous state with a new version, if you like this version better.